pytest_mh/utils/firewall.py

```diff
@@ -115,101 +257,415 @@ def teardown_when_used(self):
+            firewall.inbound.reject_port("ldap")
+            firewall.inbound.reject_port(["ldap", "ldaps"])
+
+        :param port_or_service: Port (integer), (port, protocol) tuple or service (string).
+        :type port_or_service: ServiceSpec | list[ServiceSpec]
+        """
+        self.__add_port(port_or_service, action="reject")
+
+    def drop_port(self, port_or_service: ServiceSpec | list[ServiceSpec]) -> None:  # type: ignore[override]
+        """
+        Drop incoming traffic on given port or service.
+
+        .. code-block:: python
+
+            firewall.inbound.drop_port(389)  # missing protocol defaults to "tcp"
+            firewall.inbound.drop_port((389, "tcp"))
+            firewall.inbound.drop_port("ldap")
+            firewall.inbound.drop_port(["ldap", "ldaps"])
+
+
+        :param port_or_service: Port (integer), (port, protocol) tuple or service (string).
+        :type port_or_service: ServiceSpec | list[ServiceSpec]
+        """
+        self.__add_port(port_or_service, action="drop")
+
+    def __add_port(
         self, port_or_service: ServiceSpec | list[ServiceSpec], *, action: Literal["accept", "reject", "drop"]
     ) -> None:
         def __add_port(port: int, protocol: str):
-            self.add_rich_rule(f'rule priority="{self._priority}" port port="{port}" protocol="{protocol}" {action}')
+            self.firewall.add_rich_rule(f'port port="{port}" protocol="{protocol}" {action}')
 
         def __add_service(name: str):
-            self.add_rich_rule(f'rule priority="{self._priority}" service name="{name}" {action}')
+            self.firewall.add_rich_rule(f'service name="{name}" {action}')
 
-        if not isinstance(port_or_service, list):
-            port_or_service = [port_or_service]
-
-        for item in port_or_service:
+        items = port_or_service if isinstance(port_or_service, list) else [port_or_service]
+        for item in items:
             if isinstance(item, str):
                 __add_service(item)
                 continue
 
-            port, protocol = self.parse_port_spec(item)
+            port, protocol = self.firewall.parse_port_spec(item)
             __add_port(port, protocol)
 
-        self._priority -= 1
 
+class FirewalldOutboundRules(FirewallOutboundRules):
+    """
+    Configure firewall outbound rules using firewalld.
+
+    All changes are automatically reverted when a test is finished.
+    """
+
+    def __init__(self, firewall: Firewalld) -> None:
+        """
+        :param firewall: Firewalld controller.
+        :type firewall: Firewalld
+        """
+        super().__init__()
+
+        self.firewall: Firewalld = firewall
+
+    def accept_port(self, port_or_service: ServiceSpec | list[ServiceSpec]) -> None:  # type: ignore[override]
+        """
+        Accept outgoing traffic on given port or service.
+
+        .. code-block:: python
+
+            firewall.outbound.accept_port(389)  # missing protocol defaults to "tcp"
```

Comment:
Changing my mind a little bit, I honestly think it's enough to have the one set of examples in the first function, the rest I think it's unnecessary.

Reply:
Which function do you want to have documented?

Keep in mind, that IDE shows you docstring only of the current function that you are typing. If the docstring is present only in single place, it makes sense for sphinx doc, but you will loose that information in IDE.

Reply:
Actually, it seems that this is no longer the case and help() shows the inherited docstring. I updated the patch so the documentation is only in the base class.

Reply:
Thank you.